internal/crypto/cert.go

```diff
@@ -37,32 +46,74 @@ func CNFromDeviceFingerprint(fingerprint string) (string, error) {
 	return DeviceCommonNamePrefix + fingerprint, nil
 }
 
+func CertFile(name string) string {
+	return filepath.Join(config.CertificateDir(), name+".crt")
+}
+
+func KeyFile(name string) string {
+	return filepath.Join(config.CertificateDir(), name+".key")
+}
+
 type TLSCertificateConfig oscrypto.TLSCertificateConfig
 
-type CA struct {
-	Config *TLSCertificateConfig
+type CA interface {
+	GetConfig() *TLSCertificateConfig
+	EnsureServerCertificate(certFile, keyFile string, hostnames []string, expireDays int) (*TLSCertificateConfig, bool, error)
+	MakeAndWriteServerCert(certFile, keyFile string, hostnames []string, expireDays int) (*TLSCertificateConfig, error)
+	MakeServerCert(hostnames []string, expiryDays int, fns ...CertificateExtensionFunc) (*TLSCertificateConfig, error)
+	signCertificate(template *x509.Certificate, requestKey crypto.PublicKey) (*x509.Certificate, error)
+	EnsureClientCertificate(certFile, keyFile string, subjectName string, expireDays int) (*TLSCertificateConfig, bool, error)
```

Comment:
Can we add docs to this interface please :pray:

Reply:
The interface is as ugly as it gets
It was copied 1:1 from the existing struct.
I am open to suggestions in terms of revising it and/or removing some of the functions from the public exports list.